### DISCUSSION TITLE
How to get the desired result？

### DISCUSSION THREAD

**Discussion by gitlGl** [r1chardj0n3s/parse#188] [Q&A] [answered]
![mmexport1724149280752.png](https://github.com/user-attachments/assets/e3d17483-adc6-402b-806f-56fdac017fc6)

**Comment by jenisys** [comment, marked as answer]
* Use `format_ = „{part1}_2017_{part2}_{part3}xxx.txt“` (xxx mean your Chinese letters in your original format string)
* Use typed-parameters in format, like: `format_ = „{part1:d}_2017_{part2:d}_{part3:d}xxx.txt“`

**Comment by gitlGl** [comment]
sorry,I made a mistake. Thank you for your help.
That's what I actually want .
![mmexport1724152754597.png](https://github.com/user-attachments/assets/c668677c-ae44-4f02-a9c8-ec798874a7da)

  want 
part1 000728
part2 2016
    i got it
